lib/index.d.ts

```diff
@@ -94,11 +108,31 @@ declare namespace Component {
         /** Whether to use Shadow DOM */
         useShadowDom?: boolean;
     }
+
+    interface AttrSchema {
+        /** Type of the attribute, default is 'string' */
+        type?: 'string' | 'number' | 'boolean' | 'json';
+
+        /** Default value if the attr is not defined */
+        default?: any;
+
+        /** Description of attribute, what it does e.t.c */
+        description?: string;
+    }
 }
 
 type ConfigOptions<State, AppState> = Component.ConfigOptions<State, AppState>;
 
 export class Component<State, AppState = {}> extends WebComponent {
+    /**
+     * Attributes schema that defines the component's attributes and their types
```

Comment:
probably worth being explicit that it's HTML attributes in all this documentation